python/ray/tune/search/optuna/optuna_search.py

```diff
@@ -380,7 +399,10 @@ def _setup_study(self, mode: Union[str, list]):
             self._metric = DEFAULT_METRIC
 
         pruner = ot.pruners.NopPruner()
-        storage = ot.storages.InMemoryStorage()
+        if self._storage:
+            storage = self._storage
+        else:
+            storage = ot.storages.InMemoryStorage()
```

Comment:
It is better to init `self._storage` using the argument `storage`. But not reusing the `storage` argument again and re-assign value to it using `self._storage`. Please move this part to the place where `self._storage` is initialized. And using `self._storage` in the remaining part of this module.